All features added

Aaron-RN opened the pull request:
Completed html form, Railsy Forms with #form_tag, Railsy-er Forms with #form_for and allowing the editing of users along with error detection for the forms using Rails's error explanation system.

Aaron-RN committed:
created users model and controller, also added routes to users controller methods create and new

thsvr committed:
html form

Aaron-RN committed:
finished creating html form

Aaron-RN committed:
erased error explanation from form

thsvr committed:
adding forms with form_tag

Aaron-RN committed:
created form using form_for tag

thsvr committed:
add edit action, html form

Aaron-RN committed:
added red border around forms when iput values presented errors

Aaron-RN committed:
Merge pull request #1 from Aaron-RN/users-model

created users model and controller, also added routes to users contro…

Aaron-RN committed:
Merge pull request #2 from Aaron-RN/html-form

Html form

Aaron-RN committed:
Merge pull request #3 from Aaron-RN/rails-form

adding forms with form_tag

Aaron-RN committed:
Merge pull request #4 from Aaron-RN/railsyer-form

created form using form_for tag

Aaron-RN committed:
Merge pull request #5 from Aaron-RN/editing-user

Editing user

Aaron-RN committed:
stickler error fixes (auto)

Aaron-RN committed:
fixed stickler errors

Ispirett commented on app/views/users/new.html.erb:
```diff
@@ -0,0 +1,57 @@
+<input type="submit" name="commit" value="Submit">
+
+</form>
+ <% end %>
```
[Optional ]This is nice but you could simply comment them out`  <%=# label_tag 'user_password', 'Password:'%>` the **#** after the open erb tag

Aaron-RN replied:
But if we had to do that for each line, wouldn't that go against the DRY rule?

Ispirett commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,44 @@
+  end
+
+  def create
+    # @user = User.new(username: params[:username], email: params[:email],
```
Remove commented line

Ispirett commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,44 @@
+    @user = User.new(user_params)
+
+    if @user.save
+      redirect_to users_path
```
redirect_to new_users_path

Ispirett commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,44 @@
+    @user = User.find(params[:id])
+
+    if @user.update(user_params)
+      redirect_to users_path
```
redirect_to edit_users_path

Ispirett commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,44 @@
+    if @user.update(user_params)
+      redirect_to users_path
+    else
+      render :new
```
render edit

Ispirett commented on app/views/users/new.html.erb:
```diff
@@ -0,0 +1,57 @@
+
+  <%= submit_tag "Save new user"%>
+
+ <% end %>
```
This form is returning `param is missing or the value is empty: user `

Aaron-RN replied:
That's because that form has  different params required than that of the form that is active. So that User.new(params) in the Users controller for the new method would have to change relative to that form... But since that form is inactive didn't think it mattered.

Ispirett reviewed (changes requested):
## Status: Changes required
Very good Job @Aaron-RN @thsvr, you folks did great :100:  :1st_place_medal: 

There a just a few changes to make. See below.

Aaron-RN committed:
Update new.html.erb

Changed inactive form params to the same as active form

Aaron-RN committed:
Update users_controller.rb

-Update method now renders: edit and redirects_to edit_users_path.
-Create method now redirects_to new_users_path.
-Removed commented code from Create method.

bolah2009 reviewed (approved):
## Looks good to me :+1: 

Good job implementing the suggested changes.
### :heavy_check_mark: Approved
